react-components/src/hooks/useActiveReveal3dResources.ts

```diff
@@ -1,49 +1,78 @@
 import { useFdmSdk } from '../components/RevealCanvas/SDKProvider';
 import {
   type AddModelOptions,
-  type CognitePointCloudModel,
+  type CogniteModel,
+  CognitePointCloudModel,
   type DataSourceType,
   type DMDataSourceType,
   type Image360Collection,
   isDMPointCloudModel
 } from '@cognite/reveal';
 import { getModelIdAndRevisionIdFromExternalId } from './network/getModelIdAndRevisionIdFromExternalId';
-import { type LayersUrlStateParam } from '../components';
-import { useMemo } from 'react';
+import { type LayersUrlStateParam } from '../components/RevealToolbar/LayersButton/types';
+import { createContext, useContext, useMemo } from 'react';
 import { EMPTY_ARRAY } from '../utilities/constants';
+import { PointCloudDomainObject } from '../architecture/concrete/reveal/pointCloud/PointCloudDomainObject';
+import { CadDomainObject } from '../architecture/concrete/reveal/cad/CadDomainObject';
+import { isDefined } from '../utilities/isDefined';
+import { Image360CollectionDomainObject } from '../architecture/concrete/reveal/Image360Collection/Image360CollectionDomainObject';
+import { useVisibleRevealDomainObjects } from './useVisibleRevealDomainObjects';
+
+export type UseActiveReveal3dResourcesDependencies = {
+  useVisibleRevealDomainObjects: typeof useVisibleRevealDomainObjects;
+  useFdmSdk: typeof useFdmSdk;
+};
+
+export const defaultUseActiveReveal3dResourcesDependencies = {
+  useVisibleRevealDomainObjects,
+  useFdmSdk
+};
+
+export const UseActiveReveal3dResourcesContext =
+  createContext<UseActiveReveal3dResourcesDependencies>(
+    defaultUseActiveReveal3dResourcesDependencies
+  );
 
 export const useActiveReveal3dResources = (
   layerState: LayersUrlStateParam | undefined
 ): {
   models: AddModelOptions[];
   image360Collections: Array<Image360Collection<DataSourceType>>;
 } => {
-  const models = use3dModels();
-  const image360Collections = useImage360Collections();
+  const { useVisibleRevealDomainObjects, useFdmSdk } = useContext(
+    UseActiveReveal3dResourcesContext
+  );
+
+  const visibleDomainObjects = useVisibleRevealDomainObjects();
   const fdmSdk = useFdmSdk();
 
   const visibleModels = useMemo(() => {
     if (layerState === undefined) {
       return EMPTY_ARRAY;
     }
-    return models.filter((m) => m.visible);
-  }, [models, layerState]);
 
-  const filteredImage360Collections = useMemo(() => {
+    return visibleDomainObjects
+      .filter(
+        (domainObject) =>
+          domainObject instanceof PointCloudDomainObject || domainObject instanceof CadDomainObject
+      )
+      .map((domainObject) => domainObject.model)
+      .filter(isDefined);
+  }, [visibleDomainObjects, layerState]);
+
+  const visibleImage360Collections = useMemo(() => {
     if (layerState === undefined) {
       return EMPTY_ARRAY;
     }
-    return image360Collections.filter((c) => c.getIconsVisibility());
-  }, [image360Collections, layerState]);
+    return visibleDomainObjects
```

Comment:
Why do you base visible 360 collections on `visibleDomainObjects` which are a list of CAD and Point Cloud domain objects